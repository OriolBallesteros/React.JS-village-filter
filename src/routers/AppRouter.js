import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import App from '../components/App';
import PeopleList from '../components/PeopleList';
import NotFoundPage from '../components/NotFoundPage';
import PeopleInfoItem from '../components/PeopleInfoItem';

const AppRouter = () => (
    <BrowserRouter>
        <div>
            <Switch>
                <Route path="/" component={App} exact={true} />
                <Route path="/searched" component={PeopleList} />
                <Route path="/selected/:id" component={PeopleInfoItem} />
                <Route component={NotFoundPage} />
            </Switch>
        </div>
    </BrowserRouter>
);

export default AppRouter;

