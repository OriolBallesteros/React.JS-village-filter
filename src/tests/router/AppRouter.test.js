import React from "react";
import { shallow } from "enzyme";
import AppRouter from "../../routers/AppRouter";

it("should define AppRouter correctly", () => {
    const wrapper = shallow(<AppRouter />);

    expect(wrapper).toMatchSnapshot();
});
