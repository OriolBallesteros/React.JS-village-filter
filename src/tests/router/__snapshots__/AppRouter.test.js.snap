// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should define AppRouter correctly 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/searched"
      />
      <Route
        component={[Function]}
        path="/selected/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
